rudderlabs/rudder-server
# fix(router): trying to stop the router before the generatorLoop is started causes router to hang

# Description
Stopping the router too early causes workers getting stopped properly and `Start`'s completion channel to never close.

Additional improvements to various multi-tenant related tests

**TestMultiTenant** changes:
- Spin-up a transformer at the beginning for the processor to use
- Use a unique release key in etcd for each test
- First switch to `NORMAL` mode before requesting to switch to `DEGRADED` mode
- Verify message reaches to the destination after switching to `NORMAL` mode
- Require that no errors happen when we are sending messages to the server
- Fail the test when the spawn server process fails unexpectedly
- Perform gw table verifications in real-time (stop using `require.Eventually`)

**TestDynamicClusterManager** changes:

- always teardown jobsDBs

## Notion Ticket

[Link](https://www.notion.so/rudderstacks/8aac9087df644365acdf64e28e290153?v=6e06b0a5ade24f0aa5ffe05dc2972e84&p=0e93e67e9ff3454d80c934dbe5b5f497&pm=s)

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.
